app/controllers/credentials_controller.rb

```diff
@@ -35,10 +35,18 @@ class CredentialsController < ApplicationController
   # This method requires a PUT request. The new password is in the request body.
   def update_password
     password = request.body.read
-    
-    @role.credentials.password = password
-    @role.credentials.save
-    head 204
+
+    begin
+      @role.credentials.password = password
+      @role.credentials.save(raise_on_save_failure: true)
+      head 204
+    rescue
```

Comment:
I would love to see rescue for a specific error that is thrown during save action instead of searching for message string.